src/maxminddb.c

```diff
@@ -318,7 +322,7 @@ LOCAL int map_file(MMDB_s *const mmdb)
     if (INVALID_HANDLE_VALUE != fd) {
         CloseHandle(fd);
     }
-    if (INVALID_HANDLE_VALUE != mmh) {
+    if (INVALID_HANDLE_VALUE != mmh && NULL != mmh) {
```

Comment:
This is the cleanup code that already existed - the existing code does "goto cleanup;" which is this code, and may not be visible in the snippets in the pull request.

The code changes on line 284 mean that mmh could be either INVALID_HANDLE_VALUE or NULL at this point, so we should only CloseHandle(mmh) if mmh is not either of those values.